CRUDRecipeEF.PL/Menus/IngredientMenu.cs

```diff
@@ -0,0 +1,184 @@
+                    break;
+                case IngredientMenuOption.DeleteIngredient:
+                    await DeleteIngredient();
+                    break;
+                case IngredientMenuOption.GoBack:
+                    Console.WriteLine();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private async Task LookupIngredient()
+        {
+            ConsoleHelper.ColorWrite("What ingredient would you like to lookup: ");
+            var name = Console.ReadLine();
+
+            Console.WriteLine();
+
+            try
+            {
+                var ingredient = await _ingredientService.GetIngredientByName(name);
+                ConsoleHelper.ColorWriteLine(ConsoleColor.DarkYellow, $"{name} exists.");
+            }
+            catch(KeyNotFoundException)
+            {
+                ConsoleHelper.ColorWriteLine(ConsoleColor.DarkYellow,$"{name} does not exist.");
+            }
+
+            Console.WriteLine();
+            await this.Show();
+        }
+
+        private async Task DeleteIngredient()
+        {
+            ConsoleHelper.ColorWrite("What ingredient would you like to delete: ");
+            var name = Console.ReadLine();
+
+            try
+            {
+                await _ingredientService.DeleteIngredient(name);
+            }
+            catch (KeyNotFoundException)
+            {
+                ConsoleHelper.ColorWriteLine(ConsoleColor.DarkYellow, $"{name} does not exist.");
+            }
+
+            Console.WriteLine();
+            await this.Show();
+        }
+
+        private async Task NewIngredient()
+        {
+            ConsoleHelper.ColorWrite("What ingredient would you like to add: ");
+            var name = Console.ReadLine();
+
+            IngredientAddDTO newIngreditent = new IngredientAddDTO { Name = name };
+
+            try
+            {
+                await _ingredientService.AddIngredient(newIngreditent);
+            }
+            catch (KeyNotFoundException)
+            {
+                ConsoleHelper.ColorWriteLine(ConsoleColor.DarkYellow, $"{name} already exists.");
+            }
+
+            Console.WriteLine();
+            await this.Show();
+        }
+
+        private async Task ListIngredients()
+        {
+            Console.WriteLine();
+            ConsoleHelper.ColorWriteLine("Known Ingredients: ");
+
+            var result = await _ingredientService.GetAllIngredients();
+            List<IngredientDetailDTO> ingredientList = result.ToList();
+
+            var currentIndex = 0;
+            foreach(var ingredient in ingredientList)
+            {
+                if (currentIndex % 5 == 0 && currentIndex != 0)
+                {
+                    Console.WriteLine();
+                    ConsoleHelper.ColorWriteLine(ConsoleColor.Yellow,"Press enter for next page.");
+                    Console.ReadLine();
+                }
+                currentIndex++;
```

Comment:
Use a for loop instead since you need access to the index

Reply:
Done!